phoenix-core-client/src/main/java/org/apache/phoenix/expression/util/bson/SQLComparisonExpressionUtils.java

```diff
@@ -0,0 +1,344 @@
+      expression = FIELD_EXISTS_PATTERN.matcher(expression).replaceAll(FUNC_FIELD_EXISTS);
+    }
+    if (patternsMatched.contains("=")) {
+      expression = EQUALS1_PATTERN.matcher(expression).replaceAll(FUNC_EQUALS);
+    }
+    if (patternsMatched.contains("==")) {
+      expression = EQUALS2_PATTERN.matcher(expression).replaceAll(FUNC_EQUALS);
+    }
+    if (patternsMatched.contains("!=")) {
+      expression = NOT_EQUALS1_PATTERN.matcher(expression).replaceAll(FUNC_NOT_EQUALS);
+    }
+    if (patternsMatched.contains("<>")) {
+      expression = NOT_EQUALS2_PATTERN.matcher(expression).replaceAll(FUNC_NOT_EQUALS);
+    }
+    if (patternsMatched.contains("<")) {
+      expression = LESS_THAN_PATTERN.matcher(expression).replaceAll(FUNC_LESS_THAN);
+    }
+    if (patternsMatched.contains("<=")) {
+      expression =
+              LESS_THAN_OR_EQUALS_PATTERN.matcher(expression).replaceAll(FUNC_LESS_THAN_OR_EQUALS);
+    }
+    if (patternsMatched.contains(">")) {
+      expression = GREATER_THAN_PATTERN.matcher(expression).replaceAll(FUNC_GREATER_THAN);
+    }
+    if (patternsMatched.contains(">=")) {
+      expression = GREATER_THAN_OR_EQUALS_PATTERN.matcher(expression)
+              .replaceAll(FUNC_GREATER_THAN_OR_EQUALS);
+    }
+    if (patternsMatched.contains("BETWEEN")) {
+      expression = BETWEEN_PATTERN.matcher(expression).replaceAll(FUNC_BETWEEN);
+    }
+    if (patternsMatched.contains("IN")) {
+      expression = IN_PATTERN.matcher(expression).replaceAll(FUNC_IN);
+    }
+    if (patternsMatched.contains("AND")) {
+      expression = AND_PATTERN.matcher(expression).replaceAll(OP_AND);
+    }
+    if (patternsMatched.contains("OR")) {
+      expression = OR_PATTERN.matcher(expression).replaceAll(OP_OR);
+    }
+    if (patternsMatched.contains("NOT")) {
+      expression = NOT_PATTERN.matcher(expression).replaceAll(OP_NOT);
+    }
+    return expression;
+  }
+
+  /**
+   * Returns true if the given field exists in the document.
+   *
+   * @param documentField The document field.
+   * @return True if the given field exists in the document.
+   */
+  public boolean exists(final String documentField) {
+    BsonValue topLevelValue = rawBsonDocument.get(documentField);
+    if (topLevelValue != null) {
+      return true;
+    }
+    return CommonComparisonExpressionUtils.getFieldFromDocument(documentField, rawBsonDocument)
+        != null;
+  }
+
+  /**
+   * Returns true if the value of the field is less than the value represented by {@code
+   * expectedFieldValue}. The comparison can happen only if the data type of both values match.
+   *
+   * @param fieldKey The field key for which value is compared against expectedFieldValue.
+   * @param expectedFieldValue The literal value to compare against the field value.
+   * @return True if the value of the field is less than expectedFieldValue.
+   */
+  public boolean lessThan(final String fieldKey, final String expectedFieldValue) {
```

Comment:
Yes i can make this, but since we need separate methods in SQL style comparison as this PR uses MVEL expression language and having separate function calls are clean for expression language, i kept it this way:

- SQL style expression -> contains individual comparison methods -> CommonComparisonExpressionUtils -> single method for all comparisons
- Document style expression -> contains individual comparison methods -> CommonComparisonExpressionUtils -> single method for all comparisons

When i introduce ANTLR style expression evaluation as follow up Jira, i can make this change as well:

- SQL style expression -> single method for all comparisons -> CommonComparisonExpressionUtils -> single method for all comparisons
- Document style expression -> single method for all comparisons -> CommonComparisonExpressionUtils -> single method for all comparisons

@kadirozde does this sound good?

Reply:
You can still have a separate method for each of them if you need but they should call a common method to eliminate duplicate code.

Reply:
Yes, they all call common method eventually: https://github.com/apache/phoenix/pull/1906/files#diff-d0b48fb7d6da791c1eb8a7abccad83a3e891396695be2e744983583217aeac23R154

```
  public static boolean compareValues(BsonValue value, BsonValue compareValue, CompareOp op) {
    Preconditions.checkArgument(op != null, "Comparison operator should not be null");
    if (value.isString() && compareValue.isString()) {
      int compare =
              ((BsonString) value).getValue().compareTo(((BsonString) compareValue).getValue());
      switch (op) {
        case LESS:
          return compare < 0;
        case LESS_OR_EQUAL:
          return compare <= 0;
        case GREATER:
          return compare > 0;
        case GREATER_OR_EQUAL:
          return compare >= 0;
      }
    }
...
...
```

Here, CompareOp takes the comparison operator enum value and evaluates accordingly

Reply:
This is taken care of in the latest revision.